apps/studio/src/store/modules/LicenseModule.ts

```diff
@@ -103,14 +104,45 @@ export const LicenseModule: Module<State, RootState>  = {
         await Vue.prototype.$util.send('license/createTrialLicense')
         await Vue.prototype.$noty.info("Your 14 day free trial has started, enjoy!")
       } else {
-        await Vue.prototype.$util.send('license/add', { email, key })
+        const result = await CloudClient.getLicense(window.platformInfo.cloudUrl, email, key);
+        // if we got here, license is good.
+        await Vue.prototype.$util.send('license/wipe');
+        let license = {} as TransportLicenseKey;
+        license.key = key;
+        license.email = email;
+        license.validUntil = new Date(result.validUntil);
+        license.supportUntil = new Date(result.supportUntil);
+        license.maxAllowedAppRelease = result.maxAllowedAppRelease;
+        license.licenseType = result.licenseType;
+        await Vue.prototype.$util.send('appdb/license/save', { obj: license });
       }
       // allow emitting expired license events next time
       SmartLocalStorage.setBool('expiredLicenseEventsEmitted', false)
       await context.dispatch('sync')
     },
-    async update() {
-      await Vue.prototype.$util.send('license/update')
+    async update(_context, license: TransportLicenseKey) {
+      try {
+        const data = await CloudClient.getLicense(window.platformInfo.cloudUrl, license.email, license.key)
+        license.validUntil = new Date(data.validUntil)
+        license.supportUntil = new Date(data.supportUntil)
+        license.maxAllowedAppRelease = data.maxAllowedAppRelease
+        license = await Vue.prototype.$util.send('appdb/license/save', { obj: license });
+      } catch (error) {
+        if (error instanceof CloudError) {
```

Comment:
Do we actually always get a CloudError here? I don't think we throw a cloud error on a 404. You probably need to check for a 404 specifically.